gridstatus/ieso.py

```diff
@@ -0,0 +1,307 @@
+import io
+import time
+import xml.etree.ElementTree as ET
+
+import pandas as pd
+import requests
+
+from gridstatus import utils
+from gridstatus.base import ISOBase, NotSupported
+from gridstatus.decorators import support_date_range
+from gridstatus.gs_logging import log
+
+# Load goes back several decades, with each year in one file.
+EARLIEST_LOAD = pd.Timestamp("2002-05-01").date()
+LOAD_INDEX_URL = "http://reports.ieso.ca/public/DemandZonal/"
+
+# The yearly file is updated daily, and contains data from the start of the year
+# through the current date.
+LOAD_TEMPLATE_URL = f"{LOAD_INDEX_URL}/PUB_DemandZonal_YYYY.csv"
+
+
+LOAD_FORECAST_INDEX_URL = "http://reports.ieso.ca/public/OntarioZonalDemand/"
+
+# Each forecast file contains data from the day in the filename going forward for
+# 34 days. The most recent file does not have a date in the filename.
+LOAD_FORECAST_TEMPLATE_URL = (
+    f"{LOAD_FORECAST_INDEX_URL}/PUB_OntarioZonalDemand_YYYYMMDD.xml"
+)
+
+# The farthest in the past that forecast files are available
+MAXIMUM_DAYS_IN_PAST_FOR_LOAD_FORECAST = 90
+# The farthest in the future that forecasts are available. Note that there are not
+# files for these future forecasts, they are in the current day's file.
+MAXIMUM_DAYS_IN_FUTURE_FOR_LOAD_FORECAST = 34
+
+
+INTERVAL_DURATION = pd.Timedelta(hours=1)
+
+
+class IESO(ISOBase):
+    """Independent Electricity System Operator (IESO)"""
+
+    name = "Independent Electricity System Operator"
+    iso_id = "ieso"
+
+    # All data is provided in EST, and does not change with DST. This means there are
+    # no repeated or missing hours in the raw data and we can safely use tz_localize
+    # without setting ambiguous or nonexistent times.
+    # https://www.ieso.ca/-/media/Files/IESO/Document-Library/engage/ca/ca-Introduction-to-the-Capacity-Auction.ashx
+    default_timezone = "EST"
+
+    status_homepage = "https://www.ieso.ca/en/Power-Data"
+
+    @support_date_range(frequency="YEAR_START")
+    def get_load(self, date, end=None, verbose=False):
+        """
+        Get hourly load by zone for a given date or from date to end date.
+
+        Args:
+            date (datetime.date): date to get load for. If end is None, returns
+                only data for this date.
+            end (datetime.date, optional): end date. Defaults to None. If provided,
+                returns data from date to end date.
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+            pd.DataFrame: zonal load as a wide table with columns for each zone
+        """
+        today = self._today()
+        date = self._handle_input_date(date)
```

Comment:
It may not make sense to allow `"today"` for this method sense the hourly load is only available for yesterday and earlier. (Technically, the file from yesterday contains the first hour from today but I don't think that's what callers expect). 

Then, `"latest"` would return all the data from yesterday.

Reply:
yes, "latest" should always return latest available. 

im now realizing we cant use this end point for the latest data since it's published the following day. were going to have to use whatever powers this graph

<img width="1151" alt="image" src="https://github.com/kmax12/gridstatus/assets/371792/d2448624-f36a-46cd-a961-c83bca71004d">

based on some poking around. it actually looks like this: https://www.ieso.ca/ieso/api/PowerData/DemandSupplyPrice. not the [xml](https://www.ieso.ca/-/media/files/ieso/uploaded/chart/ontario_demand_multiday.xml?la=en)) report we previously discussed

since this is the most real time dataset and is slightly different (has ontario and market demand, but no zonal), we should probably have the `get_load` method return just the ontario and market load from that file. the method would take a parameter to indicating hourly or 5min. on the hosted gridstatus api side, we'll end up implementing 2 data dataset `ieso_load_hourly` and `ieso_load_5_min` (or something like that). 

then, what you currently implemented here would be a separate `get_load_by_zone` function since the columns are different. based on my checking of last few hours of yesterday, the ontario demand from the url above appears to match the historical data in this file after rounding to nearest integer, which is good. at some point maybe we use it to backfill. it also matches the [xml file](https://www.ieso.ca/-/media/files/ieso/uploaded/chart/ontario_demand_multiday.xml?la=en) that is linked from the graph above. 
what confuses me, is the 5 minute data doesn't seem to match the other reports i see, but i am okay to ignore that for now and circle back

Reply:
Did you see my comment [here](https://github.com/kmax12/gridstatus/pull/306/files#r1436136254)? 

Where are the timestamps in the file you linked?

Reply:
By the way, did you find that API endpoint by looking at the network requests? I can't find any documentation on it, but I see it's called when the chart page loads.

Reply:
yep, that's where i found it